CIP-dapp-bridge-rework/README.md

```diff
@@ -0,0 +1,1169 @@
+
+6. **Reducing Dependencies**: Currently, all DApps developers **must** operate only with CBOR-CDDL entities, making them dependent on third-party libraries such as `cardano_serialization_lib`, and forcing them to integrate these libraries into their projects. This complicates the logic, increases the size of the loaded page, necessitates keeping up with updates, etc., thereby increasing the number of failure factors in the application's logic.<br><br>
+
+7. **Lack of abstraction**: The Cardano ecosystem is actively developing, especially in everything related to Plutus. DApps struggle to keep up with the innovations. Because of this, we encounter a heavy inconsistency in approaches. For example, this concerns everything related to collaterals at the moment. Cardano has long allowed making their use transparent and convenient for the user, but not all DApps support these approaches, forcing wallets to "reserve collateral UTxO" and somehow integrate this into UX. Moreover, the very fact that the DApp operates the collateral, and not the wallet (which only offers collateral but without guarantees that it will be used), makes the user vulnerable.<br><br>
+
+8. **Security**: Currently, DApps are given too much responsibility in the formation of transactions. This could potentially become an attack vector on a user who is confident in their security. Besides the risk of leaking wallet data, including addresses, there is a possibility that a DApp could form a malware transaction. This transaction might include fields not checked by the wallet but still obtain all necessary signatures. For example, the user could unknowingly become part of a criminal scheme or lose funds.<br><br>
+
+9. **Trustful system**: While blockchain is inherently a trustless system, it seems very odd when a wallet unconditionally trusts all user data to a third party at the first request.<br><br>
+
+By solving these problems, we will lower the entry barrier for DApp developers into the ecosystem, while simultaneously enhancing the security of the end-user who has entrusted their funds to a wallet, but should not trust their management to just any application they encounter. The wallet will act as a black box, and the DApp as an extension of its UI, not a replacement of its engine.
+
+Additionally, this approach will simplify the integration of wallets and DApps from the wallet developers' perspective, as it will provide a more natural interface for interaction, easily translatable into native scenarios of interaction with the ordinary user through the wallet's own UI.
+
+## Specification
+
+### The main idea
+The main idea is to delegate as many operations as possible to the wallet, which are its core functions. This will greatly simplify the lives of DApps developers. It will also normalize the transaction pipeline, which is currently highly intermingled between the DApp and the wallet.
```

Comment:
> This will greatly simplify the lives of DApps developers

It will only simplify the lives of developers who want to operate on a high level of abstraction

Reply:
Not only that. Even if you want to work with a lower level of abstraction, this CIP brings automation to your life. But you can still operate at a low level. You can request a draft of the transaction from the wallet for your `TransactionScript`, and then use its inputs, outputs, whatever, for your purpose to build another `TransactionScript`. You can even merge many transactions into one. You used the automation at the first stage, but then you can build a low-level `TransactionScript` if you wish. But you still don't have to write your own "semi-wallet". So even if you are a low-level developer, your life becomes easier.

That's what I was talking about - the scalability of complexity. Or "adaptive complexity". It's as high/low level as you need.